src/lib/system_logger.ts

```diff
@@ -0,0 +1,74 @@
+const systemLogTag = '__nfSystemLog'
+
+const serializeError = (error: Error): Record<string, unknown> => {
+  const cause = error?.cause instanceof Error ? serializeError(error.cause) : error.cause
+
+  return {
+    error: error.message,
+    error_cause: cause,
+    error_stack: error.stack,
+  }
+}
+
+// eslint pretends there's a different enum at the same place - it's wrong!
+// eslint-disable-next-line no-shadow
+export enum LogLevel {
+  Debug = 1,
+  Log,
+  Error,
+}
+
+class SystemLogger {
+  // eslint-disable-next-line no-useless-constructor
+  constructor(private readonly fields: Record<string, unknown> = {}, private readonly logLevel = LogLevel.Log) {}
```

Comment:
I'm confused. Don't we need to store these things in the class?

Reply:
https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties

Reply:
Interesting. To me we're left with something that is less expressive and an empty constructor that we need an ESLint exception for. 🤷🏻

Reply:
there you go: https://github.com/netlify/functions/pull/457/commits/139158e8c568bc579ac8c332e5ba62918bdfc0fa 😁